### PR TITLE
Validation cleanup parts 5 & 6 together

### PR DESCRIPTION
**Pull request by thockin**
Part 6 of several

Make a sub-pkg and shorter names for better readability
Some errors that happen during validation are not validation errors.

@lavalamp

### CODE REVIEW

**Comment by thockin** [comment]
This is the last validation PR that I have pending (though I have more that are half-cooked).

**Comment by thockin** [comment]
I'm happy to reassign this - I know you are leaving on vaca soon

**Comment by lavalamp** [comment]
Yeah, I'm really sorry-- the past few days I've been continually opening this up and then not actually reviewing it, I did it again like five minutes ago :(

**Comment by thockin** [comment]
Shooting randomly into the crowd, I hit @caesarxuchao

File: pkg/registry/service/rest.go [hunk 1]
```diff
@@ -84,15 +84,18 @@ func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 		// Allocate next available.
 		ip, err := rs.serviceIPs.AllocateNext()
 		if err != nil {
-			el := utilvalidation.ErrorList{utilvalidation.NewInvalidError(utilvalidation.NewFieldPath("spec", "clusterIP"), service.Spec.ClusterIP, err.Error())}
-			return nil, errors.NewInvalid("Service", service.Name, el)
+			// TODO: what error should be returned here?  It's not a
+			// field-level validation failure (the field is valid), and it's
+			// not really an internal error.
```
Comment (caesarxuchao) [inline pkg/registry/service/rest.go]: It seems the only possible cause of error here is the ports pool is depleted, so how about a `503 Service Unavailable`?

File: pkg/registry/pod/etcd/etcd.go [hunk 2]
```diff
@@ -134,10 +134,12 @@ func (r *BindingREST) Create(ctx api.Context, obj runtime.Object) (out runtime.O
 	binding := obj.(*api.Binding)
 	// TODO: move me to a binding strategy
 	if len(binding.Target.Kind) != 0 && binding.Target.Kind != "Node" {
-		return nil, errors.NewInvalid("binding", binding.Name, validation.ErrorList{validation.NewInvalidError(validation.NewFieldPath("target", "kind"), binding.Target.Kind, "must be empty or 'Node'")})
+		// TODO: When validation becomes versioned, this gets more complicated.
+		return nil, errors.NewInvalid("binding", binding.Name, field.ErrorList{field.NotSupported(field.NewPath("target", "kind"), binding.Target.Kind, []string{"Node", ""})})
```
Comment (caesarxuchao) [inline pkg/registry/pod/etcd/etcd.go]: I don't think people will notice the empty string in `[]string{"Node", ""}`. Just spell it out as `empty` is better IMO.

**Comment by caesarxuchao** [comment]
LGTM except a nit.

File: pkg/registry/service/rest.go (hunk 1 above)
Comment (thockin) [inline pkg/registry/service/rest.go]: I didn't want to change behavior in this PR, just cleanup.  That's a good answer for a different PR :)

File: pkg/registry/pod/etcd/etcd.go (hunk 2 above)
Comment (thockin) [inline pkg/registry/pod/etcd/etcd.go]: fixed.

**Comment by thockin** [comment]
Rebasing now

**Comment by caesarxuchao** [comment]
Thanks. LGTM.

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by thockin** [comment]
@k8s-bot unit test this please

**Pull request merged**